app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt

```diff
@@ -0,0 +1,324 @@
+    private var view: CalculatorView = mock()
+
+    @Before
+    fun setup() {
+        model = CalculatorModel()
+        presenter = CalculatorPresenter(model, view)
+    }
+
+    @Test
+    fun eraseTest() {
+        presenter.erase()
+        assertEquals(EMPTY_STRING, model.getResult())
+        verify(view).drawNumber(EMPTY_STRING)
+    }
+
+    @Test
+    fun testOperationSymbolPressedPlus() {
+        model.saveNumber(NUMBER_THREE)
+        presenter.operationSymbolPressed(OPERATOR_SUM)
+        verify(view).drawNumber(NUMBER_THREE + OPERATOR_SUM)
+        assertEquals(NUMBER_THREE + OPERATOR_SUM, model.getValue())
+    }
+
+    @Test
+    fun testOperationSymbolPressedNegative() {
+        model.saveNumber(NUMBER_THREE)//ojo aca
+        presenter.operationSymbolPressed(OPERATOR_MINUS)
+        verify(view).drawNumber(NUMBER_THREE + OPERATOR_MINUS)
+        assertEquals(NUMBER_THREE + OPERATOR_MINUS, model.getValue())
+    }
+
+    @Test
+    fun testOperationSymbolPressedDivide() {
+        model.saveNumber(NUMBER_THREE)
+        presenter.operationSymbolPressed(OPERATOR_DIVIDE)
+        verify(view).drawNumber(model.getValue())
+        assertEquals(NUMBER_THREE + OPERATOR_DIVIDE, model.getValue())
+    }
+
+    @Test
+    fun testOperationSymbolPressedWithMultiply() {
+        model.saveNumber(NUMBER_THREE)
+        presenter.operationSymbolPressed(OPERATOR_MULTIPLY)
+        verify(view).drawNumber(model.getValue())
+        assertEquals(NUMBER_THREE + OPERATOR_MULTIPLY, model.getValue())
+    }
+
+
+    @Test
+    fun testPartialDivideOperation() {
+        model.saveNumber(NUMBER_FOUR)
+        model.saveOperationSymbol(OPERATOR_DIVIDE)
+        presenter.numberPressed(NUMBER_TWO)
+        verify(view).drawNumber(model.getValue())
+        assertEquals(NUMBER_FOUR + OPERATOR_DIVIDE + NUMBER_TWO, model.getValue())
+    }
+
+    @Test
+    fun testPartialMultiplyOperation() {
+        model.saveNumber(NUMBER_TWO)
+        model.saveOperationSymbol(OPERATOR_MULTIPLY)
+        presenter.numberPressed(NUMBER_TWO)
+        verify(view).drawNumber(model.getValue())
+        assertEquals(NUMBER_TWO + OPERATOR_MULTIPLY + NUMBER_TWO, model.getValue())
+    }
+
+    @Test
+    fun testFirstOperandPressed() {
+        presenter.numberPressed(NUMBER_THREE)
+        verify(view).drawNumber(model.getValue())
+        assertEquals(NUMBER_THREE, model.getValue())
+    }
+
+    @Test
+    fun testSecondOperandPressed() {
+        model.saveNumber(NUMBER_THREE)
+        model.saveOperationSymbol(OPERATOR_SUM)
+        presenter.numberPressed(NUMBER_THREE)
+        verify(view).drawNumber(model.getValue())
+        assertEquals(NUMBER_THREE + OPERATOR_SUM + NUMBER_THREE, model.getValue())
+    }
+
+    @Test
+    fun testSecondOperandNegativePressed() {
+        model.saveNumber(NUMBER_THREE)
+        model.saveOperationSymbol(OPERATOR_SUM)
+        presenter.numberPressed(OPERATOR_MINUS + NUMBER_THREE)
```

Comment:
is it ok that the presenter is called with (-3). I think it should be called twice, first with - and then with 3

Reply:
Done thx